config/module.go

```diff
@@ -35,6 +35,8 @@ type Module struct {
 	// They overwrite existing environment variables.
 	Environment map[string]string `json:"env,omitempty"`
 
+	// Status refers to the validations done in the APP to make sure a module is configured correctly
+	Status           *appValidationStatus `json:"status"`
```

Comment:
I am slightly worried that this is a pointer. None of our other config objects contain pointers (that aren't populated inside the rdk) (that I know of). It is *probably* fine...

The reason I am worried is that it means marshalling and unmarshalling this struct will result in a different pointer, which will lead to differences. (I'm guessing this is why you set status to nil in `Equals`). 

Would it be okay to make this a non-pointer type? Then I don't think you need the `nil` sets in `Equals()`

Up to you.

Reply:
Yeah, it makes sense we don't do it for primitive data types because they will just be set to an empty string which is easy to check but for nested objects its a bit harder.